zebra-chain/src/transaction/tests/vectors.rs

```diff
@@ -37,6 +37,34 @@ fn librustzcash_tx_hash() {
     assert_eq!(hash, expected);
 }
 
+#[test]
+fn doesnt_deserialize_transaction_with_invalid_value_balance() {
+    zebra_test::init();
+
+    let dummy_transaction = Transaction::V4 {
+        inputs: vec![],
+        outputs: vec![],
+        lock_time: LockTime::Height(Height(1)),
+        expiry_height: Height(10),
+        joinsplit_data: None,
+        sapling_shielded_data: None,
+    };
+
+    let mut input_bytes = Vec::new();
+    dummy_transaction
+        .zcash_serialize(&mut input_bytes)
+        .expect("dummy transaction should serialize");
+    // Set value balance to non-zero
+    input_bytes[24] = 1;
```

Comment:
Let's modify the low-order byte, setting the value to `1`?
```suggestion
    // Set value balance to non-zero
    // There are 4 * 4 byte fields and 2 * 1 byte compact sizes = 18 bytes before the 8 byte amount
    // (Zcash is little-endian unless otherwise specified: https://zips.z.cash/protocol/nu5.pdf#endian )
    input_bytes[18] = 1;
```